### DISCUSSION TITLE
Is it possible to wait for one of multiple elements with SeleniumBase?

### DISCUSSION THREAD

**Discussion by krishpuri** [seleniumbase/SeleniumBase#2788] [Q&A] [answered]
I have a form on a webpage that can behave differently based on the input provided.

So is it possible to efficiently wait for the presence of one of multiple elements? Currently, I can achieve this without SeleniumBase by using the following code:

```
WebDriverWait(self.browser.driver, 10).until(EC.any_of(
                EC.presence_of_element_located((By.XPATH, 'elem1')),
                EC.presence_of_element_located((By.XPATH, 'elem2')),
                EC.presence_of_element_located((By.XPATH, 'elem3'))
            ))
```

And I know SeleniumBase has a wait_for_element_present() method, but is there a way to pass multiple selectors to it?

**Comment by subh-sk** [comment, marked as answer]
hii @krishpuri ,

Below is a solution that leverages the WebDriverWait and EC.any_of methods from Selenium to wait for any one of multiple elements to be present. This approach can be integrated seamlessly into a `SeleniumBase` test case.

```py
from seleniumbase import SB, BaseCase
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

def wait_for_any_element(sb: BaseCase):
    sb.driver.get("https://www.google.com")
    
    WebDriverWait(sb.driver, 10).until(EC.any_of(
                EC.presence_of_element_located((By.XPATH, 'elem1')),
                EC.presence_of_element_located((By.XPATH, 'elem2')),
                EC.presence_of_element_located((By.XPATH, 'elem3'))
            ))
    
    input("wait = ")

with SB(uc=True) as sb:
    wait_for_any_element(sb)
```

This solution provides an efficient way to wait for any of the multiple elements to be present using `SeleniumBase`, enhancing its capabilities with native Selenium methods. By incorporating WebDriverWait and EC.any_of, you can handle more complex scenarios in your web automation tasks.

**Comment by EMGSoftware** [comment]
If you want to use the available SeleniumBase detection methods you can do:
```
WebDriverWait(sb.driver, 10).until(EC.any_of(
                lambda _: bool(len(sb.find_elements('selector_elem1'))),
                lambda _: bool(len(sb.find_elements('selector_elem2'))),
                lambda _: bool(len(sb.find_elements('selector_elem3'))),
            ))
```
`find_elements()` will be called as many times as needed to find one of the elements without raising an exception.
